polygeist/lib/polygeist/Passes/ParallelLICM.cpp

```diff
@@ -384,12 +387,12 @@ void ParallelLICM::runOnOperation() {
       llvm::dbgs() << "\n";
     });
 
-    if (auto parLoop = dyn_cast<scf::ParallelOp>((Operation *)loop))
-      moveParallelLoopInvariantCode(parLoop);
-    else if (auto parLoop = dyn_cast<AffineParallelOp>((Operation *)loop))
-      moveParallelLoopInvariantCode(parLoop);
-    else
-      moveLoopInvariantCode(loop);
+    moveLoopInvariantCode(loop);
+    TypeSwitch<Operation *>((Operation *)loop)
+        .Case<scf::ParallelOp>(
+            [&](auto loop) { moveParallelLoopInvariantCode(loop); })
+        .Case<AffineParallelOp>(
+            [&](auto loop) { moveParallelLoopInvariantCode(loop); });
```

Comment:
These two do not seem equivalent to me.

Reply:
Try to compare with the original code:
```
    moveLoopInvariantCode(loopLike);
    if (auto par = dyn_cast<scf::ParallelOp>((Operation *)loopLike)) {
      moveParallelLoopInvariantCode(par);
    } else if (auto par = dyn_cast<AffineParallelOp>((Operation *)loopLike)) {
      moveParallelLoopInvariantCode(par);
    }
```

Reply:
Right. The original code called ` moveLoopInvariantCode(loopLike);` all the time. I changed the current code to do the same in response to a review comment from Whitney.